src/post_iterate.rs

```diff
@@ -0,0 +1,23 @@
+use crate::fabric::{Fabric, UniqueId};
+
+// how to do this?
+pub trait PostIterate {
+    fn post_iterate(&self, fabric: &mut Fabric);
```

Comment:
```suggestion
    fn post_iterate(self, fabric: &mut Fabric);
```
Might as well take ownership of `self` because I reckon these will all be run once.

Reply:
Post iterate is to be executed after each iteration.